src/renderer/components/repositoryListView/repositoryItem.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+	<t-flexbox align-items="center" class="welcome__repository__list__item">
+		<h6>{{ repo.name | truncateFilter(30) }}</h6>
+		<t-button margin-left="auto" @click.native="selectCurrentRepository(index)">
+			Open
+		</t-button>
+		<div
+			class="welcome__repository__list__item__settings"
+			@click="openRepositorySettings(index)"
+		>
+			<settingsIcon />
+		</div>
+	</t-flexbox>
+</template>
+
+<script>
+import TFlexbox from "../TLayouts/TFlexbox";
+import truncateFilter from "../../filters/truncate";
+import TButton from "../TButton/TButton";
+import settingsIcon from "../icon/settings";
+
+export default {
+	name: "RepositoryItem",
+	components: {
+		TFlexbox,
+		TButton,
+		settingsIcon
+	},
+	filters: { truncateFilter },
+	props: {
```

Comment:
Can you remove `repo` prop and add other props?

* `repositoryName` (String)
* `directoryPath` (String)
* `isGit` (Boolean)
* `repositoryId` (Number)

Reply:
What's `repositoryId` here exactly?

Reply:
https://github.com/gitthermal/thermal/blob/feature/switch-to-sqlite/src/renderer/main.js#L32

Reply:
Okay. How do I access that?

Reply:
Skipping this review for now (https://github.com/gitthermal/thermal/pull/3578)